### PR TITLE
fix url decode issue while hashing

### PR DESCRIPTION
**Pull request by nikhilunni511**
# Description

Ticket: [B2-2199](https://barnebys.atlassian.net/browse/B2-2199)

**Problem statement** : Catawiki redirection not working 

**Solution**: Our approach to decoding the URL involves parsing the incoming URL query parameters as JSON, then converting them into a string before generating a hash. By using this method, we ensure that the URL is properly decoded.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My changes generate no new warnings
- [ ] I have added tests for the new code/features
- [x] New and existing unit tests pass locally with my changes
- [x] No lint errors when running locally

[B2-2199]: https://barnebys.atlassian.net/browse/B2-2199?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by marcuscarlsson86 — APPROVED** [review]
This and the explanation in the ticket looks good. Approved

File: api/track.js [hunk 1]
```diff
@@ -2,6 +2,8 @@ import encodeUrl from 'encodeurl';
 import microSession from 'micro-cookie-session';
 import { send } from 'micro';
 import md5 from 'md5';
+const querystring = require("querystring");
```
Comment (MatteoCinti) [inline api/track.js:5]: why require?

Comment (nikhilunni511) [reply api/track.js:5]: 'let's just hope for good enough' over 'strive for perfection' 😉 
Will change 👍

**Comment by sony-sebastian** [comment]
https://dev.azure.com/barnebys/Barnebys%20Analytics/_git/Barnebys%20Analytics/pullrequest/14

**Pull request closed without merging**